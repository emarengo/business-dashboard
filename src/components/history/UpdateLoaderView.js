import React from 'react';

import styled from '@emotion/styled';

import { ReactComponent as NoVersionsSvg } from '../../assets/combined-shape.svg';

const UpdateLogo = styled(NoVersionsSvg)`
  margin-bottom: 30px;
`;
const UpdateLoaderContainer = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-image: linear-gradient(to bottom, #dde6f2 0%, #f2f6fa);
  color: #093548;
  font-family: 'Jost', sans-serif;
  z-index: 5;
`;

const UpdateLoaderTitle = styled.h1`
  font-size: 25px;
  font-weight: 800;
  margin-bottom: 16px;
`;

const UpdateLoaderText = styled.p`
  width: 412px;
  font-size: 17px;
  line-height: 1.31;
  font-weight: 400;
  text-align: center;
`;

const UpdateLoaderView = () => (
  <UpdateLoaderContainer>
    <UpdateLogo data-testid="UpdateLogo" />
    <UpdateLoaderTitle data-testid="UpdateLoaderTitle">
      Version still uploadingâ€¦
    </UpdateLoaderTitle>
    <UpdateLoaderText data-testid="UpdateLoaderText">
      This page is inaccessible as a new version is being deployed. <br />{' '}
      Please try again in 2 minutes.
    </UpdateLoaderText>
  </UpdateLoaderContainer>
);

export default UpdateLoaderView;
