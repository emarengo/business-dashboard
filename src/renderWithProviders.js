import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from 'emotion-theming';
import { BeatTheme } from '@thebeatapp/beat-ui';

const customRender = (node, ...options) => {
  return render(
    <Router>
      <ThemeProvider theme={BeatTheme}>{node}</ThemeProvider>
    </Router>,
    ...options
  );
};

export * from '@testing-library/react';
export { customRender as render };
export { customRender as rerender };
