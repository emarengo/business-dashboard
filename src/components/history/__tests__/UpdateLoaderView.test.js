import React from 'react';
import { cleanup, render } from '../../../renderWithProviders';
import UpdateLoaderView from '../UpdateLoaderView';

describe('Update Loader View', () => {
  let renderResult;
  let node;

  const title = 'Version still uploadingâ€¦';

  beforeEach(() => {
    renderResult = render(<UpdateLoaderView />);
  });

  afterEach(cleanup);

  it('should display the logo', () => {
    node = renderResult.getByTestId('UpdateLogo');
    expect(node).toBeInTheDocument();
  });

  it('should display title', () => {
    node = renderResult.getByTestId('UpdateLoaderTitle');
    expect(node).toHaveTextContent(title);
  });

  it('should display description text', () => {
    node = renderResult.getByTestId('UpdateLoaderText');
    expect(node).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    expect(node).toMatchSnapshot();
  });
});
