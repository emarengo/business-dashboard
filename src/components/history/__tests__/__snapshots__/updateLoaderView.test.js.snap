// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Loader View should match snapshot 1`] = `
<p
  class="css-1rfud7a"
  data-testid="UpdateLoaderText"
>
  This page is inaccessible as a new version is being deployed. 
  <br />
   
  Please try again in 2 minutes.
</p>
`;
